### ISSUE TITLE
Use hover/click on rendered keys to highlight or jump to the corresponding object in the layout JSON

### ISSUE THREAD

**Issue by nickcoutsos** [nickcoutsos/keymap-layout-tools#8]
It would help a lot with inspecting/tweaking to easily jump from the resulting key layout to its canonical definition without having to count or guess at it.

CodeMirror's `decorations` make it possible to highlight sections of text, and perhaps TreeSitter could be used to map parsed JSON code to particular objects the same way I'm doing this with tree-sitter-devicetree-viewer.

**Comment by nickcoutsos** [comment]
I went down a bit of a rabbit hole with this one trying to find suitable ways to access the tree sitter parser (which must be initialized asynchronously) in the Redux reducer via dependency injection.

The idea was to parse the layout JSON during an update and store JSON start/end indices for each key object in the state. This would make it possible to have actions that programmatically edit parts of the JSON instead of replacing it outright, but then I realized that actions like Reorder are doing full replaces anyway making it pointless.

Along the way I rewrote the reducer and used async thunks. The thunks aren't necessary anymore and just feel wrong, so I'll likely revert that part some day.

Anyway, clicking or drag selecting keys will now highlight the associated sections in the JSON panel using CodeMirrors decorations, and there's a new button to delete selected keys.
